src/main_functions.c

```diff
@@ -0,0 +1,606 @@
+				} else {
+					userconsent = 0;
+				}
+			}
+
+			char* newword = strdup(c);
+			printf("\nReplacing \"%s\" with \"%s\". \n", misspelled[i], newword);
+			correct_line(line_copy, misspelled[i], newword); //modifies line function
+
+         	printf("%s", line_copy);
+        	/* if the line is the last line being edited, the last character is
+        	 * EOF, not \n. */
+         	if (returnflag) printf("\n");
+         	printf("%s", underline_misspelled_sentence(misspelled, line_copy, underline));
+
+        } else if (isdigit(choice[0]) && (atoi(&choice[0]) <= max_no_suggestions)) { // choose suggestion
+        	// Replace misspelled word with chosen replacement
+        	int c = atoi(&choice[0]) - 1;
+        	printf("\nReplacing %s with %s.\n\n", misspelled[i], suggestions[c]);
+        	correct_line(line_copy, misspelled[i], suggestions[c]); //modifies line function
+
+        	/* if the line is the last line being edited, the last character is
+        	 * EOF, not \n. */
+         	printf("%s", line_copy);
+         	if (returnflag) printf("\n");
+         	printf("%s", underline_misspelled_sentence(misspelled, line_copy, underline));
+        }
+
+        i++;	// added loop changer
+    }
+
+	return line_copy;
+}
+
+
+char** interactive_mode(char* filename, dict_t* dict, int* quit)
+{
+	char** lines;
+	lines = parse_file(filename);
+
+	// step through phases
+	int i = 0;
+	int flag = 0;
+	int linenumber;
+
+	while (lines[i] != NULL) {
+		linenumber = i+1;
+		if (lines[i+1] == NULL) flag = 1;	// last line
+		lines[i] = edit_interactive(lines[i], dict, linenumber, flag);
+		i++;
+	}
+
+    *quit = 0;
+	return lines;
+}
+/* 
+	IV. Batch Mode
+ */
+// Adapted from sarika's edit_interactive
+// returns corrections to a list that contains misspelled words, or returns a corrected char** string
+// that can represent an array
+char* edit_batch(char* line, dict_t* dict, int verbosity, int lnum)
+{
+
+	char *line_copy = strdup(line);
+    int max_no_suggestions = 2; //should the user decide this?
+    int length = strlen(line);
+    char **misspelled; //generates an empty array where the misspelled words in a line will be stored
+
+    misspelled = calloc(length, sizeof(char*));
+    if (misspelled == NULL) {
+    	fprintf(stderr,"edit_interactive: calloc failed");
+    	exit(0);
+    }
+
+    char *underline = (char *)malloc(sizeof(char) * strlen(line + 1)); //generate an empty array where the underline will go
+    underline[0] = '\0';
+
+    parse_string(line, dict, underline, misspelled); //identify misspelled words and add to misspelled
+    char *suggestions[max_no_suggestions]; //generates empty array where suggestions will be filled
+
+    int i = 0;
+    //replacing words, printing out if batch mode
+    while (misspelled[i] != NULL) {
+        int rc = generate_suggestions(misspelled[i], dict, suggestions);
+
+	    if (rc == EXIT_FAILURE) {
+            if (verbosity) suggestions[0] = "No suggestions generated"; 
```

Comment:
What the heck? Why are we overwriting the closest suggestions if we are being verbose?

Reply:
rc == EXIT_FAILURE is only if no suggestions were generated in the first place, so the message that needs to be printed out is no suggestions generated.